variable/to_unit.cpp

```diff
@@ -16,8 +16,9 @@ namespace {
 constexpr double days_multiplier = llnl::units::precise::day.multiplier();
 }
 
-Variable to_unit(const Variable &var, const units::Unit &unit) {
-  if (unit == var.unit())
+Variable to_unit(const Variable &var, const units::Unit &unit,
+                 const CopyPolicy copy) {
+  if (unit == var.unit() && copy == CopyPolicy::TryAvoid)
     return var; // no copy
```

Comment:
Not sure it makes a difference, but could consider using:
```suggestion
  if (unit == var.unit())
    return copy == CopyPolicy::Always ? copy(var) : var;
```